src/resources/stores.jl

```diff
@@ -21,43 +20,47 @@ See [`Container`](@ref) for a more lock-like resource.
 
 Think of `Resource` and `Container` as locks and of `Store` as channels. They block only if empty (on taking) or full (on storing).
 """
-mutable struct Store{T} <: AbstractResource
+mutable struct Store{N, T<:Number} <: AbstractResource
   env :: Environment
   capacity :: UInt
   load :: UInt
-  items :: Dict{T, UInt}
+  items :: Dict{N, UInt}
   seid :: UInt
-  put_queue :: DataStructures.PriorityQueue{Put, StorePutKey{T}}
-  get_queue :: DataStructures.PriorityQueue{Get, StoreGetKey}
-  function Store{T}(env::Environment; capacity=typemax(UInt)) where {T}
-    new(env, UInt(capacity), zero(UInt), Dict{T, UInt}(), zero(UInt), DataStructures.PriorityQueue{Put, StorePutKey{T}}(), DataStructures.PriorityQueue{Get, StoreGetKey}())
+  put_queue :: DataStructures.PriorityQueue{Put, StorePutKey{N, T}}
+  get_queue :: DataStructures.PriorityQueue{Get, StoreGetKey{T}}
+  function Store{N, T}(env::Environment; capacity::UInt=typemax(UInt)) where {N, T<:Number}
+    new(env, capacity, zero(UInt), Dict{N, UInt}(), zero(UInt), DataStructures.PriorityQueue{Put, StorePutKey{N, T}}(), DataStructures.PriorityQueue{Get, StoreGetKey{T}}())
   end
 end
 
+function Store{N}(env::Environment; capacity::UInt=typemax(UInt)) where {N}
+  Store{N, Int}(env; capacity)
+end    
+    
 """
     put!(sto::Store, item::T)
 
 Put an item into the store. Returns the put event, blocking if the store is full.
 """
-function put!(sto::Store{T}, item::T; priority::Int=0) where T
```

Comment:
This does not need to be modified, but just FYI, the `::T` in the priority kwarg (and the previously existing `::Int`) are unnecessary (if I understand Julia's dispatch rules correctly). Basically, type signatures of keyword arguments are not used for dispatch, so they are frequently superfluous.

Reply:
They are still useful as a type assert though

Reply:
we now convert to the appropriate type given the type of the Store